internal/controllers/v1beta1/gateway/gateway.go

```diff
@@ -105,6 +103,11 @@ func (c *GatewayController) Reconcile(ctx context.Context, request reconcile.Req
 		}, err
 	}
 
+	//If we don't have the tls management label or it isn't set to true return
+	if val, ok := gateway.Labels[v1beta1labels.InjectSimpleCredentialNameLabel]; !ok || val != "true" {
+		return reconcile.Result{}, nil
+	}
```

Comment:
This is the inverse of the check in `admission.go`.  A function can be created clean it up.